sycl/doc/design/OffloadDesign.md

```diff
@@ -0,0 +1,396 @@
+# Implementation design for offloading model
+
+## Introduction
+
+This document covers the implementation design for using the new offloading
```

Comment:
Not sure if we should call it 'implementation design'. May be 'proposed design'?